codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -168,30 +170,91 @@ private static void encode3to4(
         // significant bytes passed in the array.
         // We have to shift left 24 in order to flush out the 1's that appear
         // when Java treats a value as negative that is cast from a byte to an int.
-        int inBuff =
-                (numSigBytes > 0? src.getByte(srcOffset)     << 24 >>>  8 : 0) |
-                (numSigBytes > 1? src.getByte(srcOffset + 1) << 24 >>> 16 : 0) |
-                (numSigBytes > 2? src.getByte(srcOffset + 2) << 24 >>> 24 : 0);
+        if (dest.order() == ByteOrder.BIG_ENDIAN) {
+            final int inBuff;
+            if (numSigBytes <= 0) {
+                inBuff = 0;
+            } else if (numSigBytes == 1) {
+                inBuff = src.getByte(srcOffset) << 24 >>> 8;
+            } else if (numSigBytes == 2) {
+                short value = src.getShort(srcOffset);
+                inBuff = ((byte) (value >>> 8)) << 24 >>> 8 |
+                         ((byte) value) << 24 >>> 16;
+            } else {
+                int value = src.getMedium(srcOffset);
+                inBuff = ((byte) (value >>> 16)) << 24 >>> 8 |
+                         ((byte) (value >>> 8)) << 24 >>> 16 |
+                         ((byte) value) << 24 >>> 24;
+            }
+            encode3to4BigEndian(inBuff, numSigBytes, dest, destOffset, alphabet);
+        } else {
+            final int inBuff;
+            if (numSigBytes <= 0) {
+                inBuff = 0;
+            } else if (numSigBytes == 1) {
+                inBuff = src.getByte(srcOffset)  << 24 >>>  8;
+            } else if (numSigBytes == 2) {
+                short value = src.getShort(srcOffset);
+                inBuff = ((byte) value) << 24 >>>  8 |
+                         ((byte) (value >>> 8)) << 24 >>> 16;
+            } else {
+                int value = src.getMedium(srcOffset);
+                inBuff = ((byte) value) << 24 >>> 8          |
+                         ((byte) (value >>> 8)) << 24 >>> 16 |
+                         ((byte) (value >>> 16)) << 24 >>> 24;
+            }
 
+            encode3to4LittleEndian(inBuff, numSigBytes, dest, destOffset, alphabet);
+        }
+    }
+
+    private static void encode3to4BigEndian(
+            int inBuff, int numSigBytes, ByteBuf dest, int destOffset, byte[] alphabet) {
+        // Packing bytes into an int to reduce bound and reference count checking.
+        switch (numSigBytes) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)